matrix/Dcm.hpp

```diff
@@ -28,20 +34,68 @@ class Dcm : public Matrix<Type, 3, 3>
 
     typedef Matrix<Type, 3, 1> Vector3;
 
+    /**
+     * Standard constructor
+     *
+     * Initializes to identity
+     */
     Dcm() : Matrix<Type, 3, 3>()
     {
         (*this) = eye<Type, 3>();
     }
 
+    /**
+     * Constructor from array
+     *
+     * @param other array
+     */
     Dcm(const Type *data_) : Matrix<Type, 3, 3>(data_)
     {
     }
 
+    /**
+     * Copy constructor
+     *
+     * @param other Matrix33 to set dcm to
+     */
     Dcm(const Matrix<Type, 3, 3> &other) : Matrix<Type, 3, 3>(other)
     {
     }
 
+    /**
+     * Constructor from quaternion
+     *
+     * Instance is initialized from quaternion representing
+     * transformation from inertial frame to body frame.
+     *
+     * @param q quaternion to set dcm to
+     */
     Dcm(const Quaternion<Type> & q) {
+        set_from_quaternion(q);
+    }
+
+    /**
+     * Constructor from euler angles
+     *
+     * Instance is initialized from angle tripplet (3,2,1) representing
+     * transformation from body frame to inertial frame. 
+     *
+     * @param euler euler angles to set dcm to
+     */
+    Dcm(const Euler<Type> & euler) {
+        set_from_euler(euler);
+    }
+
+
+    /**
+     * Set from quaternion
+     *
+     * Instance is set from quaternion representing
+     * transformation from inertial frame to body frame.
```

Comment:
The terms earth and body are very application specific. I recommend referring the starting frame as 'frame 1' and the finishing frame as 'frame 2'